include/xtensor/xmanipulation.hpp

```diff
@@ -608,6 +608,32 @@ namespace xt
 
         return rot90_impl<n>()(std::forward<E>(e), norm_axes);
     }
+
+    template<class E>
+    inline auto roll(E&& e, std::ptrdiff_t shift) {
+      decltype(auto) cpy = empty_like(std::forward<E>(e));
+      auto eflat_view = flatten(e);
+      auto cpyflat_view = flatten(cpy);
+      if(shift == 0) {
+        std::copy(eflat_view.begin(), eflat_view.end(), cpyflat_view.begin());
+      }
+      else if(shift < 0) {
+        shift = -shift;
+        if(shift > eflat_view.size())
+          shift %= eflat_view.size();
```

Comment:
Formatting: single statement in a conditional branch should be surrounded with curly braces.